### ISSUE TITLE
Unnatural behavior when focusing on CheckBox.Item with screen reader

### ISSUE THREAD

**Issue by grgr-dkrk** [callstack/react-native-paper#2920]
### Current behaviour
The behavior of `CheckBox.Item` in screen readers differs between OSs, both of which lack context such as state.
 
On Android, focus on the checkbox label and each checkbox. They are read aloud separately and may lack the context of checkboxes.
The state of the checkbox can be changed by double tapping on the label component. However, it lacks information such as `accessibilityRole`.

On iOS, it focus only on checkbox labels. The child checkbox is not focused and the state is not read aloud.
Like Android, The state of the checkbox can be changed by double tapping on the label component. However, it does not announce the state of the checkbox.

Please check out the "Screenshots" video for more details.

### Expected behaviour
We need to tell the screen reader the label of the checkbox, the status, and that they are the checkbox itself.

### Code sample
https://github.com/callstack/react-native-paper/blob/main/example/src/Examples/CheckboxItemExample.tsx

#### Android
https://user-images.githubusercontent.com/40130327/136792062-64eed816-0ed9-4dca-9be9-b263635b915a.mp4

#### iOS
https://user-images.githubusercontent.com/40130327/136794945-d6ff5541-819f-49c3-8249-ccd18c30aef4.mp4

### What have you tried
At least in the current version of React Native (0.65), we can't associate elements on Android. iOS also cannot focus on child elements of components that have `accessible`.

Thus, I tried to deal with it by setting the role, label and state in the parent element `CheckBox.Item`.
Please check my PR(#2921) for details.

### Your Environment

| software              | version
| --------------------- | -------
| ios or android        | iOS14, Android11
| react-native          | 0.63
| react-native-paper    | 4.9.2
| node                  | 16.11.0
| npm or yarn           | 1.22.10
| expo sdk              | 40.0.0
